stdlib/LinearAlgebra/src/lu.jl

```diff
@@ -527,7 +527,14 @@ function lu!(A::Tridiagonal{T,V}, pivot::Union{RowMaximum,NoPivot} = RowMaximum(
     if dl === du
         throw(ArgumentError("off-diagonals of `A` must not alias"))
     end
-    du2 = fill!(similar(d, max(0, n-2)), 0)::V
+    # Check if Tridiagonal matrix already has du2 for pivoting
+    has_du2_defined = isdefined(A, :du2) && isa(A.du2, V) && length(A.du2) == max(0, n-2)
```

Comment:
Seems like if `du2` is defined, then it has to be of type `V` by the parametrization, though the length check is necessary, see lines 500ff in `tridiag.jl`.

Reply:
Good call